### PR TITLE
Bug multigpu trade fixed.

### PR DESCRIPTION
**Pull request by VahidooX**

### CODE REVIEW

File: nemo/collections/nlp/callbacks/state_tracking_trade_callback.py [hunk 1]
```diff
@@ -33,27 +34,39 @@ def eval_iter_callback(tensors, global_vars, data_desc):
...
-    global_vars['comp_res'].extend(comp_res.cpu().numpy())
-    global_vars['gating_preds'].extend(torch.argmax(gate_outputs, axis=-1).cpu().numpy())
+    point_outputs_max_list = []
+    tgt_ids_list = []
+    gate_outputs_max_list = []
+    for tensor_name, values_list in tensors.items():
+        if tensor_name.startswith('loss'):
+            for values in values_list:
+                global_vars['loss'].append(tensor2list(values))
+        if tensor_name.startswith('gating_labels'):
```
Comment (ekmb) [inline nemo/collections/nlp/callbacks/state_tracking_trade_callback.py:44]: this and the following could be `elif`.

File: nemo/collections/nlp/callbacks/state_tracking_trade_callback.py [hunk 2]
```diff
@@ -33,27 +34,39 @@ def eval_iter_callback(tensors, global_vars, data_desc):
...
-    comp_res = (point_outputs_max == tgt_ids) | mask_paddings
-    comp_res = torch.all(comp_res, axis=-1, keepdims=False)
-
-    global_vars['comp_res'].extend(comp_res.cpu().numpy())
-    global_vars['gating_preds'].extend(torch.argmax(gate_outputs, axis=-1).cpu().numpy())
+    point_outputs_max_list = []
+    tgt_ids_list = []
+    gate_outputs_max_list = []
+    for tensor_name, values_list in tensors.items():
+        if tensor_name.startswith('loss'):
```
Comment (ekmb) [inline nemo/collections/nlp/callbacks/state_tracking_trade_callback.py:41]: you don't use `global_vars[loss]` in the epoch_done(), why do you need it here?

Comment (VahidooX) [reply nemo/collections/nlp/callbacks/state_tracking_trade_callback.py:41]: It was there for debugging to have the loss for evaluation. I would remove it.

File: nemo/collections/nlp/callbacks/state_tracking_trade_callback.py (hunk 1 above)
Comment (VahidooX) [reply nemo/collections/nlp/callbacks/state_tracking_trade_callback.py:44]: Done.

**Pull request merged**